fedimint-client/src/module/mod.rs

```diff
@@ -102,6 +102,87 @@ pub trait ClientModule: Debug + MaybeSend + MaybeSync + 'static {
     ) -> anyhow::Result<()> {
         anyhow::bail!("Wiping not supported");
     }
+
+    /// Does this module support being a primary module
+    ///
+    /// If it does it must implement:
+    ///
+    /// * [`Self::create_sufficient_input`]
+    /// * [`Self::create_exact_output`]
+    /// * [`Self::await_primary_module_output`]
+    /// * [`Self::get_balance`]
+    /// * [`Self::subscribe_balance_changes`]
+    fn supports_being_primary(&self) -> bool {
+        false
+    }
+
+    // TODO: unclear if we should return a vec of inputs
```

Comment:
I discussed something related with @joschisan: if we ever want to make e-cash notes separate inputs/outputs and remove `TieredMulti` that would be useful.